internal/querynodev2/services.go

```diff
@@ -753,69 +753,41 @@ func (node *QueryNode) Search(ctx context.Context, req *querypb.SearchRequest) (
 		return resp, nil
 	}
 
-	toReduceResults := make([]*internalpb.SearchResults, len(req.GetDmlChannels()))
-	runningGp, runningCtx := errgroup.WithContext(ctx)
-
-	for i, ch := range req.GetDmlChannels() {
-		ch := ch
-		req := &querypb.SearchRequest{
-			Req:             req.Req,
-			DmlChannels:     []string{ch},
-			SegmentIDs:      req.SegmentIDs,
-			Scope:           req.Scope,
-			TotalChannelNum: req.TotalChannelNum,
-		}
-
-		i := i
-		runningGp.Go(func() error {
-			ret, err := node.searchChannel(runningCtx, req, ch)
-			if err != nil {
-				return err
-			}
-			if err := merr.Error(ret.GetStatus()); err != nil {
-				return err
-			}
-			toReduceResults[i] = ret
-			return nil
-		})
-	}
-	if err := runningGp.Wait(); err != nil {
+	if len(req.GetDmlChannels()) != 1 {
+		err := merr.WrapErrParameterInvalid(1, len(req.GetDmlChannels()), "count of channel to be searched should only be 1, wrong code")
 		resp.Status = merr.Status(err)
+		log.Warn("got wrong number of channels to be searched", zap.Error(err))
 		return resp, nil
 	}
 
-	tr.RecordSpan()
-	var result *internalpb.SearchResults
-	var err2 error
-	if req.GetReq().GetIsAdvanced() {
-		result, err2 = segments.ReduceAdvancedSearchResults(ctx, toReduceResults, req.Req.GetNq())
-	} else {
-		result, err2 = segments.ReduceSearchResults(ctx, toReduceResults, segments.NewReduceInfo(req.Req.GetNq(),
-			req.Req.GetTopk(),
-			req.Req.GetExtraSearchParam().GetGroupByFieldId(),
-			req.Req.GetExtraSearchParam().GetGroupSize(),
```

Comment:
currently in milvus query framework, only one channel will be searched, queried in one request, which is triggered by proxy, So I simplify code here and remove unnessary reduction between different channels which will never happen at all